### ISSUE TITLE
Hass.io does not run after PI reboot 

### ISSUE THREAD

**Issue by greencharlie100** [home-assistant/hassio#869]
Using:
Raspberry PI, Hass.io
Connected to Wi-Fi using USB stick named: 'CONFIG'

I have Hassio running for a number of weeks, however, if the PI shuts down/reboots due to 'me' unplugging/replugging the Pi power cable - Hassio will not start up afterwards.

I am trying to produce a robust system and am simulating an electrical power failure by unplugging/replugging the PI. 

I thought it was an issue with the PI, so I bought a new one. same problem.

The only way I get Hassio running again is to re-flash the OS and start from scratch.

Additional Information - when PI reboots:
Hass.io does not run NOR does the PI connect to my network 
Is Hassio not booting because it stalls because it can't establish a network connection?
Or
Is a network connection not established simply because Hass.io doesn't run and everything else stalls behind it?

Appreciate the help...

**Comment by frenck** [comment]
A Pi run of an SD card... and are not resistant to power surges. So unplugging the power suddenly is a recipe for a broken system in many cases.

This is not something Hassio can do anything about, this is just the Raspberry Pi.

**Comment by greencharlie100** [comment]
@frenck that is incorrect. 

I have several Raspberry PI's operating for this past 18 months and they will always boot no matter what I do to them. I could pull the power cable 100 times and the OS will run and all my services will auto start as normal. 

A sudden power failure is something that everyone will experience now and again. Are you telling me Hass.io cannot recover from this? 

I am not the first person to rais this issue but no one has addressed it properly or attempted to solve it.

I would hope that a solution is available or people will look for alternatives to hass.io.

**Comment by arontsang** [comment]
@greencharlie100

Frenck is correct. The issue is hardware. It is impossible to correct without switching to a read only file system, which is impractical in most cases.

Google about for "raspberry pi SD corrupt" and you will see a plathora of resources on the subject.

I would put money in the 100s on times you power cycled your pi was during idle. Try the same experiment with Raspbian whilst you are doing many random disk write/erase. 

The issue is due to Flash memory physics, SD card Garbage Collection and I/O scheduling.

You can fix this by adding a UPS for the pi to cleanly complete GC before power failure, or switching to a more reliable storage device, such as HDD or possibly an external SSD (which will have capacitors to safely shutdown).

**Comment by greencharlie100** [comment]
@arontsang, I can only go by what I have been experiencing with my previous systems running on Raspberry Pi's. I am new to HA and Hassio. I made the choice to dive into Hass.io and spent a few weeks building my Home system, to later discover the booting/startup issue after a power failure.

I ditched Hass.io this past couple of days and went for the raw Hassbian OS instead. A lot of time wasted on Hass.io but I learned from my mistake.

Turns out I can pull the power on Hassbian (HA) and the system boots every time. This is how all systems are designed. SD cars are permeant storage. Systems are designed to recover. Hass.io is not. The issue is with Hass.io not designed for disaster recovery.

Hassio running on Docker containers seems like and advancement but it’s like creating a 3-wheel car - its new, more efficient in many ways but it’s NOT BETTER.

Hass.io is great in many ways, but its going at a sprinting pace before it can walk properly.
Build disaster recovery into Hass.io and it can be great. Otherwise each and every user has to reflash the OS every time there is an unexpected power failure.
